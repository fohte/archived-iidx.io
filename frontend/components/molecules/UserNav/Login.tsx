import { Button } from 'antd'
import Link from 'next/link'
import * as React from 'react'

const Login: React.SFC = () => (
  <Link href="/login" prefetch>
    <Button>Login</Button>
  </Link>
)

export default Login
