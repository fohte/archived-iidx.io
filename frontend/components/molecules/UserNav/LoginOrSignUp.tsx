import { Button } from 'antd'
import Link from 'next/link'

const LoginOrSignUp: React.SFC = () => (
  <Link href="/login" prefetch>
    <Button>Login / SignUp</Button>
  </Link>
)

export default LoginOrSignUp
