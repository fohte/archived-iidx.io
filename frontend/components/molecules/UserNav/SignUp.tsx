import { Button } from 'antd'
import Link from 'next/link'

const SignUp: React.SFC = () => (
  <Link href="/signup" prefetch>
    <Button>SignUp</Button>
  </Link>
)

export default SignUp
