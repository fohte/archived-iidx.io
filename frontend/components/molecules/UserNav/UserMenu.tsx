import { auth } from 'lib/firebaseApp'

export interface Props {
  displayName: string
}

const UserMenu: React.SFC<Props> = ({ displayName }) => (
  <div>
    {displayName}{' '}
    <button
      onClick={async () => {
        await auth.signOut()
      }}
    >
      signout
    </button>
  </div>
)

export default UserMenu
