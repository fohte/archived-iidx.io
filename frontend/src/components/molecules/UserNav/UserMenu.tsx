import * as React from 'react'
import { Dropdown } from 'semantic-ui-react'

import { auth } from '@app/lib/firebaseApp'
import { Link } from '@app/routes'

export interface Props {
  displayName: string
}

const UserMenu: React.SFC<Props> = ({ displayName }) => (
  <Dropdown item text={displayName} pointing>
    <Dropdown.Menu>
      <Link route={`/@${displayName}`}>
        <Dropdown.Item>Profile</Dropdown.Item>
      </Link>
      <Dropdown.Divider />
      <Link route="/results/new">
        <Dropdown.Item>Register results</Dropdown.Item>
      </Link>
      <Dropdown.Divider />
      <Dropdown.Item
        onClick={async () => {
          await auth.signOut()
        }}
      >
        Sign out
      </Dropdown.Item>
    </Dropdown.Menu>
  </Dropdown>
)

export default UserMenu
