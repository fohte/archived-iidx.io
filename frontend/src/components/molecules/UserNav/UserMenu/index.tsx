import * as React from 'react'

import { auth } from '@app/lib/firebaseApp'
import { Link } from '@app/routes'
import * as css from './style.scss'

export interface Props {
  displayName: string
}

const UserMenu: React.SFC<Props> = ({ displayName }) => (
  <div className={css.box}>
    {displayName}
    <div className={css.dropdown}>
      <Link route={`/@${displayName}`}>
        <a>
          <div className={css.item}>Profile</div>
        </a>
      </Link>
      <div className={css.divider} />
      <Link route="/results/new">
        <a>
          <div className={css.item}>Register results</div>
        </a>
      </Link>
      <div className={css.divider} />
      <div
        className={css.item}
        onClick={async () => {
          await auth.signOut()
        }}
      >
        Sign out
      </div>
    </div>
  </div>
)

export default UserMenu
