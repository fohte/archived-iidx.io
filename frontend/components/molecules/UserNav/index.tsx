import { Icon } from 'antd'
import * as React from 'react'

import { InjectedProps } from '@app/lib/withAuthState'
import LoginOrSignUp from './LoginOrSignUp'
import UserMenu from './UserMenu'

export interface Props extends InjectedProps {}

const UserNav: React.SFC<Props> = ({ loading, viewer }) => {
  if (loading) {
    return <Icon type="loading" />
  }

  if (viewer) {
    return <UserMenu displayName={viewer.uid} />
  } else {
    return <LoginOrSignUp />
  }
}

export default UserNav
