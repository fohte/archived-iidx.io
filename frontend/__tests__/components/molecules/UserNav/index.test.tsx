import * as React from 'react'
import * as renderer from 'react-test-renderer'

import UserNav from '@app/components/molecules/UserNav'

describe('UserNav', () => {
  it('renders a loading icon when the loading prop is true', () => {
    const component = renderer.create(
      <UserNav loading viewer={null} signedIn={false} />,
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders a UserMenu component', () => {
    const component = renderer.create(
      <UserNav
        loading={false}
        viewer={{ __typename: 'User', id: 'id', name: 'foo' }}
        signedIn
      />,
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders a Login component and a SignUp component', () => {
    const component = renderer.create(
      <UserNav loading={false} viewer={null} signedIn={false} />,
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
