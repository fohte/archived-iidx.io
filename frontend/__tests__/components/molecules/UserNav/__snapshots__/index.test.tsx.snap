// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNav renders a Login component and a SignUp component 1`] = `
Array [
  <button
    className="ui inverted button"
    onClick={[Function]}
  >
    SignUp
  </button>,
  <button
    className="ui inverted button"
    onClick={[Function]}
    style={
      Object {
        "marginLeft": "0.5em",
      }
    }
  >
    Login
  </button>,
]
`;

exports[`UserNav renders a UserMenu component 1`] = `
<div
  aria-expanded={false}
  className="ui item pointing dropdown"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  role="listbox"
  tabIndex={0}
>
  <div
    aria-live="polite"
    className="text"
    role="alert"
  >
    foo
  </div>
  <i
    aria-hidden="true"
    className="dropdown icon"
    onClick={[Function]}
  />
  <div
    className="menu transition"
  >
    <div
      className="item"
      onClick={[Function]}
      role="option"
    >
      Profile
    </div>
    <div
      className="divider"
    />
    <div
      className="item"
      onClick={[Function]}
      role="option"
    >
      Register results
    </div>
    <div
      className="divider"
    />
    <div
      className="item"
      onClick={[Function]}
      role="option"
    >
      Sign out
    </div>
  </div>
</div>
`;

exports[`UserNav renders a loading icon when the loading prop is true 1`] = `
<div
  className="ui tiny active inline loader"
/>
`;
